### PR TITLE
Start drafting documentation

### PR DESCRIPTION
**Pull request by deccer**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,94 +1,217 @@
...
+}
+```
+
+Now let me explain what all these things mean.
+
+```cpp
+#include <GLFW/glfw3.h>
+```
+
+C++ needs to know where all the definitions and declarations are coming from.
```
Comment (16-Bit-Dog) [inline docs/index.md:137]: "C++ needs to know where all the definitions and declarations are coming from." is ambiguous and does not really specifically explain the reason of #include<GLFW/glfw3.h> specifically, I recommend modification of this phrase to:

"C++ needs to know where all the definitions and declarations of GLFW are coming from; `#include<GLFW/glfw3.h>` allows you to include the file that contains all of these definitions for GLFW."

Comment (deccer) [reply docs/index.md:137]: thats why there is a ```todo box right above the code under 0. Initial Setup

File: docs/index.md [hunk 2]
```diff
@@ -1,94 +1,217 @@
...
+
+```cpp
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+```
+
+Pretty obvious, right? Try to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further.
```
Comment (16-Bit-Dog) [inline docs/index.md:147]: recommended phrase change to: 
"Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further."
 
This is recommended because when reading the phrase the first time around, it can be very convoluted for what 
Pretty obvious, right? Try to initialize `GLFW`." is referring to; does it mean inside glfwInit, was something missed? - the small amount of extra context clears up any confusion with the statment for what is pretty obvious. You also could just say: "The following code is pretty obvious, right? ..." <-- this has the same effect but keeps the light hearted tone

Comment (deccer) [reply docs/index.md:147]: can you provide the actual markdown aleady? then i dont have to pick it out myself from your suggestion

File: docs/index.md [hunk 3]
```diff
@@ -1,94 +1,217 @@
...
+
+`GLFW` was initially meant to support development of OpenGL based applications, hence the gl in its name, but over the years it also supports other APIs not just OpenGL. Now since `GLFW` automatically create a context for OpenGL we can tell it not to do it via that window hint.
+
+```cpp
+    const int32_t windowLeft = videoMode->width / 2 - width / 2;
+    const int32_t windowTop = videoMode->height / 2 - height / 2;
+    glfwSetWindowPos(window, windowLeft, windowTop);
+```
+
+Should be fairly obvious. We like centered windows.
```
Comment (16-Bit-Dog) [inline docs/index.md:176]: I recommend to explain what glfwSetWindowPos does a bit more... why does it center a window?

"Should be fairly obvious. We like centered windows — and glfwSetWindowPos allow us to specify a window to set location to.
in this case — `window` is the GLFW window we created. Since top left of a window is the position of the window we set the window top left position by getting half the monitor size subtracted by the window we created size, getting us windowLeft and windowTop respectivly"

File: docs/index.md (hunk 2 above)
Comment (16-Bit-Dog) [reply docs/index.md:147]: sure, I will do it after - just need to sort through everything first (plus I also need to justify my suggestion - else its crass of me to suggest something arbitrarily 'just because'

File: docs/index.md (hunk 3 above)
Comment (deccer) [reply docs/index.md:176]: Do we really have to dumb it down that much?

Comment (16-Bit-Dog) [reply docs/index.md:176]: true, but else we ask the reader to search through the code to find what "videoMode" is; plus top left of window being position is not intuitive unless you previously worked with GUI windowing before.

plus we also say things overly simplified such as: "Now we clean up any resource we created, such as the window itself and the GLFW system. Then simply return to the OS, without any error."

File: docs/index.md (hunk 2 above)
Comment (16-Bit-Dog) [reply docs/index.md:147]: Mark down:  
Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further."

File: docs/index.md (hunk 1 above)
Comment (16-Bit-Dog) [reply docs/index.md:137]: ah, okay, sounds good

File: docs/index.md (hunk 2 above)
Comment (deccer) [reply docs/index.md:147]: > Mark down: `` Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further." ``

If i am not blind, thats exactly what I have written :) so not sure what you want me to change now

Comment (16-Bit-Dog) [reply docs/index.md:147]: 🤷

Comment (deccer) [reply docs/index.md:147]: I was asking for the exact markdown you want me the thing to change, but you provided my initial text in the markdown, not sure if that was your intention?!?

Comment (16-Bit-Dog) [reply docs/index.md:147]: my change is minor, it just adds the `glfwInit()` section for clarity - i'm sorry for being very unclear

File: docs/index.md [hunk 4]
```diff
@@ -1,94 +1,217 @@
...
+- `ID3D11RasterizerState` elaborate... asd
 
-        Some information inside this tab
+### Libraries & Project Layout
 
-    ```cs
-    class SomeCSharpCodeHere
+For the LearnD3D11 tutorial series we will be using the following third party libraries
+
+- [GLFW](https://www.glfw.org) It provides the window to render our things in, as well as input handling via mouse and keyboard. Yes there are also alternatives like `SDL2` for instance. It serves the same purpose, we just decided to go with `GLFW`. You might also wonder about "I want to do it myself, from scratch" then you really could do it yourself; but its the year 2022 by the time we are writing this thing and the problem of creating a window and handling input has been solved many times already, why reinvent the wheel. We have a link [here](DIYWinApi.md) to show how you could do it).
```
Comment (spnda) [inline docs/index.md:53]: For readability of the raw Markdown, please split this into multiple lines. Otherwise one has to scroll horizontally for quite a while.

**Review by spnda — COMMENTED** [review]
Just some stuft I noticed

File: docs/index.md [hunk 5]
```diff
@@ -1,94 +1,217 @@
...
+
+- [DirectXMath](...link...) As part of the Windows SDK will be the math library we use
```
[the remark this answers is not in the document]
Comment (deccer) [reply docs/index.md:55]: Sounds better

File: docs/index.md (hunk 4 above)
Comment (deccer) [reply docs/index.md:53]: You are welcome to suggest explicitly already

Comment (spnda) [reply docs/index.md:53]: I'm on my phone currently, but probably manually adding linebreaks at ~80/100 characters should be sufficient. The content itself seems good to me.

File: docs/index.md [hunk 6]
```diff
@@ -1,94 +1,217 @@
...
+
+int main(int argc, char* argv[])
+{
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+
+    const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
```
Comment (AlexB8675) [inline docs/index.md:94]: Should not be `const`

File: docs/index.md [hunk 7]
```diff
@@ -1,94 +1,217 @@
...
+    const GLFWvidMode* videoMode = glfwGetVideoMode(primaryMonitor);
```
Comment (AlexB8675) [inline docs/index.md:95]: Stray capital "M" in `GLFWvidMode`, should be: `GLFWvidmode`

File: docs/index.md [hunk 8]
```diff
@@ -1,94 +1,217 @@
...
+    const int32_t width = static_cast<int32_t>(videoMode->width * 0.9f);
+    const int32_t height = static_cast<int32_t>(videoMode->height * 0.9f);
+
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+    const GLFWwindow* window = glfwCreateWindow(
```
Comment (AlexB8675) [inline docs/index.md:101]: Should not be `const`

**Review by AlexB8675 — COMMENTED** [review]
Small syntax errors

File: docs/index.md (hunk 6 above)
Comment (deccer) [reply docs/index.md:94]: Hmm, dumb question: why not?

Comment (AlexB8675) [reply docs/index.md:94]: Because most GLFW functions require a non-const pointer to `GLFWmonitor`, for example `glfwGetVideoMode`

Comment (deccer) [reply docs/index.md:94]: hmm but it compiles fine with const in that constellation. And once the thing is moved into Application it will just be a constless field... Very possible i simply dont understand the thing.

Comment (AlexB8675) [reply docs/index.md:94]: `const auto primaryMonitor = glfwGetPrimaryMonitor();` worked fine because the type of `primaryMonitor` was of `GLFWmonitor* const`, meanwhile `const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();` fails in `glfwGetVideoMode()` since its type is now `const GLFWmonitor*`.

Comment (deccer) [reply docs/index.md:94]: Ah yes, i didnt see the star on monitor :)

File: docs/index.md (hunk 2 above)
Comment (deccer) [reply docs/index.md:147]: OK I also found the text you were referring to :)
